src/services/item/repository.ts

```diff
@@ -234,15 +235,29 @@ export const ItemRepository = AppDataSource.getRepository(Item).extend({
       .getRawOne();
   },
 
-  async getOwn(memberId: string): Promise<Item[]> {
-    return this.createQueryBuilder('item')
+  async getOwn(
+    memberId: string,
+    page: number,
+    name: string,
+    all: boolean,
+    limit: number,
+  ): Promise<Paginated<Item>> {
+    const pageLimit = limit || ITEMS_LIST_LIMIT;
+    const skip = (page - 1) * pageLimit;
+    const [data, totalCount] = await this.createQueryBuilder('item')
       .leftJoinAndSelect('item.creator', 'creator')
       .innerJoin('item_membership', 'im', 'im.item_path @> item.path')
       .where('creator.id = :id', { id: memberId })
       .andWhere('im.permission = :permission', { permission: PermissionLevel.Admin })
       .andWhere('nlevel(item.path) = 1')
+      .andWhere("LOWER(item.name) LIKE '%' || :name || '%'", {
```

Comment:
I guess this still work well when `name` is empty? Is this a filter?

Reply:
@morganridel 
yes it works well if the name is an empty string, and maybe the naming is misconceptual but yes name should be as an item name search query